src/net-ipfs-engine/Catalyst.Ipfs.Http.Server/HttpApi/V0/SwamController.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Catalyst.Ipfs.Core.CoreApi;
+using Lib.P2P;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Catalyst.Ipfs.Http.Server.HttpApi.V0
+{
+    /// <summary>
+    ///   Addresses for peers.
+    /// </summary>
+    public class AddrsDto
+    {
+        /// <summary>
+        ///   Addresses for peers.
+        /// </summary>
+        public Dictionary<string, List<string>> Addrs = new Dictionary<string, List<string>>();
+    }
+
+    /// <summary>
+    ///   Information on a peer.
+    /// </summary>
+    public class ConnectedPeerDto
+    {
+        /// <summary>
+        ///  The unique ID of the peer.
+        /// </summary>
+        public string Peer;
+
+        /// <summary>
+        ///   The connected address.
+        /// </summary>
+        public string Addr;
+
+        /// <summary>
+        ///   Avg time to the peer.
+        /// </summary>
+        public string Latency;
+
+        /// <summary>
+        ///   Creates a new peer info.
+        /// </summary>
+        public ConnectedPeerDto(Peer peer)
+        {
+            Peer = peer.Id.ToString();
+            Addr = peer.ConnectedAddress?.WithoutPeerId().ToString();
+            Latency = peer.Latency == null ? "n/a" : Duration.Stringify(peer.Latency.Value, string.Empty);
+        }
+    }
+
+    /// <summary>
+    ///   Information on a peer.
+    /// </summary>
+    public class ConnectedPeersDto
+    {
+        /// <summary>
+        ///   The connected peers.
+        /// </summary>
+        public IEnumerable<ConnectedPeerDto> Peers;
+    }
+
+    /// <summary>
+    ///   A list of filters.
+    /// </summary>
+    public class FiltersDto
+    {
+        /// <summary>
+        ///   A list of multiaddresses.
+        /// </summary>
+        public string[] Strings;
+    }
+
+    /// <summary>
+    ///   TODO
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)